Fix user profile flaky tests.

cahrens opened the pull request:
## [TNL-2704](https://openedx.atlassian.net/browse/TNL-2704)

Trying to read the value right after setting it in the bio field is flaky-- but nobody is using the return value anyway! There is a test that verifies that the bio field is properly set, but it uses a different method to get the value.

Related to TNL-2704, TNL-3492, TNL-3514, and TNL-2199.

Note that it doesn't actually address the failure stack trace in TNL-2199 (that's the only one that is different), but the test point associated with TNL-2199 has now run 30 times without failures in Jenkins, so I'm going to remove the flaky decorator with this PR and close TNL-2199 as Not Reproducible.

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @andy-armstrong 
- [x] Code review: @robrap 

FYI: @benpatterson 

### Post-review
- [ ] Squash commits

cahrens commented:
jenkins run bokchoy

andy-armstrong commented:
:+1: That's much cleaner to understand as well as being less flaky.

robrap commented on common/test/acceptance/pages/lms/fields.py:
```diff
@@ -163,26 +163,19 @@ def value_for_text_field(self, field_id, value=None, press_enter=True):
-        if value is not None:
-            query.fill(value)
-            query.results[0].send_keys(u'\ue004')  # Focus Out using TAB
+            return
```
Is it up to the caller to verify that it was set properly?  Is it a standard to quietly fail?

I get that you just dropped the return value, and didn't add this if, but just curious.

cahrens commented:
@robrap Good point-- I changed the code to wait for the element to be present. If it never becomes present, it times out (after 5 seconds I think) with this error:

1) ERROR: test_user_can_see_error_for_file_size_below_the_min_limit (common.test.acceptance.tests.lms.test_learner_profile.OwnLearnerProfilePageTest)

   Traceback (most recent call last):
    common/test/acceptance/tests/lms/test_learner_profile.py line 543 in test_user_can_see_error_for_file_size_below_the_min_limit
      profile_page = self.visit_profile_page(username, privacy=self.PRIVACY_PUBLIC)
    common/test/acceptance/tests/lms/test_learner_profile.py line 66 in visit_profile_page
      self.set_public_profile_fields_data(profile_page)
    common/test/acceptance/tests/lms/test_learner_profile.py line 50 in set_public_profile_fields_data
      profile_page.set_value_for_textarea_field('bio', 'Nothing Special')
    common/test/acceptance/pages/lms/fields.py line 174 in set_value_for_textarea_field
      self.wait_for_element_presence(field_selector, 'Editable textarea is present.')
    /edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/bok_choy/page_object.py line 509 in wait_for_element_presence
      self.wait_for(lambda: self.q(css=element_selector).present, description=description, timeout=timeout)
    /edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/bok_choy/page_object.py line 490 in wait_for
      return EmptyPromise(promise_check_func, description, timeout=timeout).fulfill()
    /edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/bok_choy/promise.py line 97 in fulfill
      raise BrokenPromise(self)
   BrokenPromise: Promise not satisfied: Editable textarea is present.

I updated the code and squashed at the same time, since the changes in this PR are so minimal.

(note that the way the code was written is a general pattern in these tests, but I think it is beyond the scope of this PR to update all of them)

benpatterson commented on common/test/acceptance/pages/lms/fields.py:
```diff
@@ -163,26 +163,19 @@ def value_for_text_field(self, field_id, value=None, press_enter=True):
+    def set_value_for_textarea_field(self, field_id, value):
         """
-        Get or set the value of a textarea field.
+        Set the value of a textarea field.
```
THANK YOU for fixing this. Having a combined getter/setter is good programming, but not deterministic enough for testing purposes. I've looked at this code a few times but never seemed to have enough time or context to upgrade it.

robrap commented:
:+1:
